qualtran/bloqs/basic_gates/z_basis.py

```diff
@@ -384,12 +384,16 @@ def check(self, attribute, val):
         if val >= 2**self.bitsize:
             raise ValueError(f"`val` is too big for bitsize {self.bitsize}")
 
+    @cached_property
+    def dtype(self) -> QDType:
+        if self.bitsize == 1:
+            return QBit()
+        return QAny(self.bitsize)
```

Comment:
Is this change supposed to be part of this PR? Looks unrelated.

Reply:
I just pulled out the dtype to avoid repeating the code (in signature and build_composite_bloq)